beacon-chain/state/state-native/setters_consolidation.go

```diff
@@ -0,0 +1,83 @@
+package state_native
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state/state-native/types"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// AppendPendingConsolidation is a mutating call to the beacon state which appends the provided
+// pending consolidation to the end of the slice on the state. This method requires access to the
+// Lock on the state and only applies in electra or later.
+func (b *BeaconState) AppendPendingConsolidation(val *ethpb.PendingConsolidation) error {
+	if b.version < version.Electra {
+		return errNotSupported("AppendPendingConsolidation", b.version)
+	}
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	b.sharedFieldReferences[types.PendingConsolidations].MinusRef()
+	b.sharedFieldReferences[types.PendingConsolidations] = stateutil.NewRef(1)
```

Comment:
Same comment here. Perhaps I'm not understanding these two lines

Reply:
Addressed in your other comment